api/STRUCTURE.md

```diff
@@ -0,0 +1,83 @@
+
+We generally name database objects in lower_snake_case.
+
+Liquibase migrations are applied (indirectly) via a bash script called [db/run-migrations.sh](https://github.com/all-of-us/workbench/blob/master/api/db/run-migrations.sh). Do not run this script directly; instead, call one of the following `project.rb` commands:
+
+- dev-up (starts a local dev server)
+- run-local-all-migrations (runs migrations on the `workbench` and `cdr` schemas)
+- run-local-rw-migrations (runs migrations only on the `workbench` schema)
+
+## DAOs
+
+Data Access Objects (DAOs) are interfaces that describe what operations the application can perform on the database. Generaly, we give DAOs the CamelCased name of the database table. All of the Workbench DAOs implement Spring [`CrudRepository`](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html), which means they can do simple operations like find-by-primary-key, save, delete-by-primary-key, etc.
+
+CrudRepository also allows queries to be [automatically derived](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details) from the name of a method defined on the interface. As such, you will see methods like `findByUserIdAndWorkspaceId` with no apparent implementation. The implementation is generated by Spring.
+
+To define a query that uses a foreign key, there are a couple of options. First, you could use `@JoinColumn`/`@[One|Many]ToOne` annotation on the corresponding DB Model, as described [below](#db-models). Second, you could use an [@Query](https://www.baeldung.com/spring-data-jpa-query) annotation to just describe the query in stringy SQL.
+
+TODO: do we have a preference of which of those to do?
+
+## DB Models
+
+DB Models represent a single row of a single table in the database, possibly with extender tables attached. We denote DB models by prefixing `Db` to the name of the database table. DB models heavily use `javax.persistence` annotations to describe where in the database table each part of the model lives.
+
+In order to automatically gather data from extender tables, we use the `@JoinColumn`, `@OneToOne`, `@OneToMany`, `@ManyToOne`, and `@MappedBy` [annotations](https://www.baeldung.com/jpa-join-column). When using these annotations, be sure to carefully consider which table should 'own' the relationship. With a `@ManyToOne` relationship, the `@ManyToOne` side should generally own the relationship. For example. if we have an extender table containing a user's addresses, `DbAddress` should have a `@ManyToOne` `DbUser` member and `DbUser` should have a `@OneToMany` `DbAddress` member, with the annotation containing a `mappedBy` attribute pointing to `DbAddress`'s `DbUser` member as the member defining the relationship.
+
+TODO: wording on the above is Bad. pls help
```

Comment:
You're getting into the weeds a bit here. Maybe breaking out a separate doc on spring data will help.

Links to usage in the code (though fragile) could help too. Might be better to put snippets of canonical usage here.